### DISCUSSION TITLE
Do http datasources support POST calls, or only GET?

### DISCUSSION THREAD

**Discussion by richgerrard** [hairyhenderson/gomplate#1138] [Q&A]
Hi!
I had an API change from a GET call to a POST call in a new release, and my datasource fails as follows:

```error calling ds: Couldn't read datasource 't': Unexpected HTTP status 404 on GET from https://t.x.y/api/v1/daemonset.yaml?type=application/yaml: 404 page not found\n"```

Is this configurable?  Docs mentioned headers, but not methods.
Thank you!

**Comment by hairyhenderson** [comment]
Hey @richgerrard! 👋

That's a pain. POST isn't supported - generally because it tends to model a write, and gomplate datasources are all about reading.

Is this particular POST looking for a body to be sent, or can it be empty? If so I could probably add support pretty easily.

One thing you could try is setting the `X-HTTP-Method-Override` header to `POST` (i.e. `--datasource-header t="X-HTTP-Method-Override=POST"`) - some proxies & APIs support that. But it's a long shot.

**Reply by richgerrard** [reply]
Thanks!  Will give that a go first!

**Reply by hairyhenderson** [reply]
@richgerrard just curious if this worked for you? If so could you mark it as answered? 🙂
